### PR TITLE
planner: guard max_min_eliminate against empty scalar aggregations

### LINKED ISSUE #69379: planner: max_min_eliminate panics on empty scalar aggregation produced by SEMI_JOIN_REWRITE
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

A deterministic reproducer at the optimizer-rule seam is:

```go
sctx := coretestsdk.MockContext()
agg := logicalop.LogicalAggregation{}.Init(sctx, 0)
_, _, _ = (&rule.MaxMinEliminator{}).Optimize(context.Background(), agg)
```

On current master without the guard below, the call panics with:

```text
panic: runtime error: index out of range [0] with length 0
```

This empty scalar aggregation shape is expected to be reachable from `SEMI_JOIN_REWRITE` on `EXISTS` subqueries, for example queries shaped like:

```sql
select *
from t
where exists (
  select /*+ SEMI_JOIN_REWRITE() */ 1
  from t as x
  where x.a < 1
);
```

### 2. What did you expect to see? (Required)

`max_min_eliminate` should treat an empty scalar aggregation as non-applicable input and return it unchanged, without panicking.

### 3. What did you see instead (Required)

`composeAggsByInnerJoin()` assumes `aggs[0]` exists. When `LogicalAggregation` has `AggFuncs=[]` and `GroupByItems=[]`, `splitAggFuncAndCheckIndices()` returns `(empty slice, true)`, and `max_min_eliminate` panics with `index out of range [0] with length 0`.

### 4. What is your TiDB version? (Required)

Current local master (branch `fix/max-min-eliminate-empty-aggs-panic`, based on a worktree that was `master...origin/master [ahead 10706]` before branching).

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: close #69379

Problem Summary: `max_min_eliminate` may panic with `index out of range [0] with length 0` when it receives an empty scalar aggregation (`AggFuncs=[]`, `GroupByItems=[]`). This shape is expected to be reachable from `SEMI_JOIN_REWRITE` on some `EXISTS` subqueries. The rule currently assumes at least one aggregate function exists and dereferences `aggs[0]`, which turns this invalid intermediate shape into an optimizer crash.

### What changed and how does it work?
- Add an early return in `max_min_eliminate` for empty scalar aggregations, so the rule treats them as non-applicable input and returns them unchanged.
- Avoid passing an empty aggregate-function slice into `composeAggsByInnerJoin()`, which previously accessed `aggs[0]` without a length check.
- Add a rule-level regression test that constructs an empty `LogicalAggregation` directly and verifies `Optimize()` does not panic and keeps the original aggregation node unchanged.

### Check List
Tests 
- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation
- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.
```release-note
Fix a panic in `max_min_eliminate` when the optimizer processes empty scalar aggregations.
```

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed handling of empty scalar aggregations during query optimization to prevent errors and ensure stable processing of edge-case queries.

### CODE REVIEW

File: pkg/planner/core/rule/rule_max_min_eliminate_test.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// Copyright 2026 PingCAP, Inc.
```
Comment (Reminiscent) [inline pkg/planner/core/rule/rule_max_min_eliminate_test.go:1]: Please add a SQL test
```
create table t(a int);
insert into t values (0), (1);

set @@tidb_opt_enable_semi_join_rewrite = on;

with c as (select * from t)
select max(t.a)
from t
where exists (
  select /*+ SEMI_JOIN_REWRITE() */ 1
  from c
  where c.a < 1
)
and exists (
  select 1
  from c
  where c.a < 2
);
```

Rest LGTM.

**Review by Reminiscent — APPROVED** [review]

File: pkg/planner/core/rule/rule_max_min_eliminate_test.go (hunk 1 above)
Comment (lichunzhu) [reply pkg/planner/core/rule/rule_max_min_eliminate_test.go:1]: addressed in https://github.com/pingcap/tidb/commit/b0ddde9ff1951a6024fb3eb3788f9f7720b02dbb

**Review by qw4990 — APPROVED** [review]

**Review by Reminiscent — APPROVED** [review]
